test_vm/src/lib.rs

```diff
@@ -211,6 +218,35 @@ impl<'bs> VM<'bs> {
         v
     }
 
+    pub fn with_epoch(self, epoch: ChainEpoch) -> VM<'bs> {
+        self.checkpoint();
+        VM {
+            store: self.store,
+            state_root: RefCell::new(self.state_root.take()),
+            actors_dirty: RefCell::new(false),
+            actors_cache: RefCell::new(HashMap::new()),
+            empty_obj_cid: self.empty_obj_cid,
+            network_version: NetworkVersion::V16,
+            curr_epoch: epoch,
+            invocations: RefCell::new(vec![]),
+        }
+    }
+
+    pub fn get_miner_balance(&self, maddr: Address) -> MinerBalances {
+        let a = self.get_actor(maddr).unwrap();
+        let st = self.get_state::<MinerState>(maddr).unwrap();
+        MinerBalances {
+            available_balance: a.balance
+                - (st.pre_commit_deposits.clone()
```

Comment:
I believe you could get rid of cloning if you'd borrow those BigInts, what do you think?

Reply:
Good call, thanks to the above review I'm delegating to the state method (which borrows)